mono/mini/method-to-ir.c

```diff
@@ -8126,36 +8181,41 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 		case CEE_LDARG_1:
 		case CEE_LDARG_2:
 		case CEE_LDARG_3:
+			n = (*ip) - CEE_LDARG_0;
+ldarg:
 			CHECK_STACK_OVF (1);
-			n = (*ip)-CEE_LDARG_0;
 			CHECK_ARG (n);
-			if (ip + 1 < end && is_addressable_valuetype_load (cfg, ip + 1, cfg->arg_types[n])) {
+			if (next_ip < end && is_addressable_valuetype_load (cfg, next_ip, cfg->arg_types[n])) {
 				EMIT_NEW_ARGLOADA (cfg, ins, n);
 			} else {
 				EMIT_NEW_ARGLOAD (cfg, ins, n);
 			}
 			*sp++ = ins;
 			break;
+
 		case CEE_LDLOC_0:
 		case CEE_LDLOC_1:
 		case CEE_LDLOC_2:
 		case CEE_LDLOC_3:
+			n = (*ip) - CEE_LDLOC_0;
+ldloc:
```

Comment:
I'm not a fan of local labels inside switch cases. Maybe we can later move this out to a function?

Reply:
Yes probably every label can be a function. Want that now?
But I don't see the problem. goto and labels are a basic feature.

Reply:
Actually the labels can probably be removed by first switching on operand type in the opcode table. Then the labels will all be adjacent.